packages/flutter_tools/lib/src/build_system/targets/custom_devices.dart

```diff
@@ -0,0 +1,234 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../artifacts.dart';
+import '../../base/file_system.dart';
+import '../../build_info.dart';
+import '../../devfs.dart';
+import '../../project.dart';
+import '../build_system.dart';
+import '../depfile.dart';
+import '../exceptions.dart';
+import 'assets.dart';
+import 'common.dart';
+import 'desktop.dart';
+import 'icon_tree_shaker.dart';
+
+/// The only files/subdirectories we care out.
+const List<String> _kLinuxArtifacts = <String>[
+  'libflutter_linux_gtk.so',
```

Comment:
Now I'm super confused. This PR doesn't actually seem to be about generic support for profile builds for custom devices, but rather it's about something more specific. I think we need to take a step back from this PR, and at least I would like to have a better understanding of the problem being solved here. Sorry if that's already in an issue somewhere and I just missed it.

Reply:
Sorry this is my mistake I copied the linux target , but did not modify them to the right extent. I will look into this.